### DISCUSSION TITLE
Appearance of code block headings in the navigation sidebar

### DISCUSSION THREAD

**Discussion by NikosAlexandris** [pawamoy/markdown-exec#31] [Q&A] [answered]
I am trying to get the same beautifully looking sections in the navigation sidebar (left) as in the documentation of this very plugin : [markdown_exec/](https://pawamoy.github.io/markdown-exec/reference/markdown_exec/) 😄 : 

![2024-01-12T16:03:40,220039700+02:00](https://github.com/pawamoy/markdown-exec/assets/7046639/604f005e-1a71-4f46-a5ec-2aa2abc5dbfe)

While there is an option for the headings to appear in the table of contents, I cannot find any documentation about the way this is done for the navigation sidebar, as seen in the [API reference](https://pawamoy.github.io/markdown-exec/reference/markdown_exec/) page.  I can only trace and guess that this is generated via the [gen_ref_nav.py](https://github.com/pawamoy/markdown-exec/blob/main/scripts/gen_ref_nav.py) script. Unless I miss something obvious, would it be considerable to expose this functionality directly to users of [markdown-exec](https://github.com/pawamoy/markdown-exec) ?

**Comment by pawamoy** [comment, marked as answer]
Good investigation :smile: 

mkdocstrings now offers this feature (symbol types in headings and in the table of contents) for free (previously it was an Insiders feature), but the *navigation* is built using the `gen_ref_nav.py` script, as you mention. Since mkdocstrings has zero control over which files are created, or their names, this feature (adding symbol types to the *navigation*) cannot be exposed, neither through mkdocstrings, nor markdown-exec.

If you want to add these symbols to your own navigation, you can follow what I do in `gen_ref_nav.py` (look for `mod_symbol`). I'm afraid there's no clean way to add such symbols in physical pages (non-generated ones) without mkdocs-literate-nav.

In the future, I want to create a new plugin that will ease the auto-generation of pages, and it will support symbols in the navigation too :slightly_smiling_face:

**Reply by NikosAlexandris** [reply]
(I can _verify_ that I did look after the Discussions page and did not find it 😆)

**Reply by pawamoy** [reply]
Yup, I got confused when I didn't find it myself, it wasn't enabled, I just enabled it :smile:
